### PR TITLE
explain why one cannot divide comptime_float by comptime_int #2568

### PR DESCRIPTION
**Pull request by MadMax129**
I have added into the Type Coercion section in the documentation the example about this issue, I can add more tests or more documentation if needed.

### CODE REVIEW

**Comment by MadMax129** [comment]
My question is if float widening is allowed in zig why would precision change not be allowed between values. Since the value and type of the float and int are known at compile-time shouldn't the compiler know if type coercion is possible or not based on the sizes of the values.

File: doc/langref.html.in [hunk 1]
```diff
@@ -4990,6 +4990,22 @@ test "float widening" {
...
     assert(d == a);
+}
+      {#code_end#}
+      {#header_close#}
+      {#header_open|Type Coercion: Coercion Float to Int#}
+      <p>
+      A compiler error is appropriate because this ambiguous expression leaves the compiler
+      two choices about the coercion.
+      <ul>
+        <li> Casting 54.0 to comptime_int(54) resulting in comptime_int(10), which is casted to f32(10)</li>
```
Comment (daurnimator) [inline doc/langref.html.in:5001]: `comptime_int(10)`/`f32(10)` are no longer valid syntax

File: doc/langref.html.in [hunk 2]
```diff
@@ -4990,6 +4990,22 @@ test "float widening" {
...
+        <li> Casting 5 to comptime_float(5.0) resulting in comptime_float(10.8), which is casted to f32(10.8)</li>
+      </ul>
+      </p>
+      {#code_begin|exe_err#}
```
Comment (andrewrk) [inline doc/langref.html.in:5005]:
```suggestion
      {#code_begin|test_err#}
```

**Comment by Vexu** [comment]
Messed up pushing some syntax fixes, landed in 50139aa23245c7b6b6a5faff95f1a2c854051a38.

**Pull request closed without merging**